agent/main.go

```diff
@@ -162,7 +162,7 @@ func newCmd() *cobra.Command {
 				resync <- true
 			})
 			go func() {
-				errors.CheckError(http.ListenAndServe(fmt.Sprintf("0.0.0.0:%d", port), nil))
+				errors.CheckErrorWithCode(http.ListenAndServe(fmt.Sprintf("0.0.0.0:%d", port), nil), errors.ErrorConnectionFailure)
```

Comment:
I think this is not a `ErrorConnectionFailure` class of error, since it creates a listener and does not try to establish a connection to the API.